### DISCUSSION TITLE
Guidelines - Block quantity and complexity

### DISCUSSION THREAD

**Discussion by layout-lab** [area17/twill#1023] [Q&A]
Hi,

I am working on a site where an page item may have 10+ blocks. 
These blocks range in complexity from simple text inputs for a heading/paragraph through to repeaters for an image carousel. On this quantity the page save action causes the browser to hang, and in some cases the order of repeater items have not saved correctly.

Are there any guidelines for:
- Max quantity of blocks per item?
- Complexity of the blocks (fields, repeaters)?

Additionally, has anyone experienced repeater items not save in the correct order?

Thanks!

**Comment by daun** [comment]
We've run into the same issue, where saving or switching the language crashes the browser. In that case, the page has about 40 block items with a repeater of 3 items inside.

Would love to get some input from the Twill team about the performance to be expected and ways to optimize block composition.

**Comment by ifox** [comment]
Hi @layout-lab @daun, sorry to be so late to address your question here, we're finally catching up on older issues and leveraging GitHub Discussions for those types of questions now.

Is this an issue that you've had to deal with a lot? We've not experienced this even though we've really created complex blocks and repeaters configuration and our clients are really pushing the system sometimes without running into freezes or crashes. I've seen Twill UI hanging a little bit when opening the content editor a few times, but not what you describe. I'm not too surprised about the implementation having it's limits, though. Ultimately this is a Vue app with Vuex and there are probably ways to optimize but it might also just run into limitations of the framework and browsers.

If I can provide one recommendation when running into this, it would be to think about why you ended up with this many blocks created on a single form. Could the editor be better served by a nested module? Could the entity be split into multiple modules referenced via browser fields?

Please feel free to share the blocks and repeaters that created those issues for you so that we can try to reproduce.
